common/ta_errors.h

```diff
@@ -179,6 +180,30 @@ typedef enum {
   SC_UTILS_TIMER_ERROR = 0x02 | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
   /**< Errors occurred in timer function */
   SC_UTILS_TIMER_EXPIRED = 0x03 | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
+  /**< Failed to send message */
+  SC_UTILS_HTTPS_SEND_ERROR = 0x04 | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
+  /**< HTTPS module initialize error */
+  SC_UTILS_HTTPS_INIT_ERROR = 0x05 | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
+  /**< HTTPS X509 certificate parse error */
+  SC_UTILS_HTTPS_X509_ERROR = 0x06 | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
+  /**< HTTPS initial connection error */
+  SC_UTILS_HTTPS_CONN_ERROR = 0x07 | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
+  /**< HTTPS setting SSL config error */
+  SC_UTILS_HTTPS_SSL_ERROR = 0x08 | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
+  /**< HTTPS response error */
+  SC_UTILS_HTTPS_RESPONSE_ERROR = 0x09 | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
+  /**< Error occured when serialize message */
+  SC_UTILS_TEXT_SERIALIZE = 0x0A | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
+  /**< Error occured when deserialize message */
+  SC_UTILS_TEXT_DESERIALIZE = 0x0B | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
+  /**< Out of memory error */
+  SC_UTILS_OOM_ERROR = 0x0C | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
+  /**< Overflow error */
+  SC_UTILS_OVERFLOW_ERROR = 0x0D | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
+  /**< Error occured when encrypt message */
+  SC_UTILS_ENCRYPT_ERROR = 0x0E | SC_MODULE_UTILS | SC_SEVERITY_FATAL,
```

Comment:
We can combine `SC_UTILS_ENCRYPT_ERROR` and `SC_UTILS_DECRYPT_ERROR` into single `SC_UTILS_CIPHER_ERROR` since we always know the operations.